profile-diff/local-utils.js

```diff
@@ -0,0 +1,100 @@
+const getProfileDiff = async (id) => {
+  try {
+    const finalUrl = `${API_BASE_URL}/profileDiffs/${id}`;
+    const res = await fetch(finalUrl, {
+      credentials: 'include',
+      method: 'GET',
+      headers: {
+        'Content-type': 'application/json',
+      },
+    });
+    if (res.status === 401) {
+      return { notAuthorized: true };
+    }
+    if (res.status === 404) {
+      return { profileDiffExist: false };
+    }
+    return { profileDiffExist: true, ...(await res.json()).profileDiff };
```

Comment:
- do we have a common function which is used to make api calls and handles throwing error for 4xx and 5xx response? if so please let's use it.

- if not I would suggest you make a very thin abstraction on fetch and use it here, because in the current case we're not handling 5xx or any other 4xx response.

Reply:
We do not have that function as of now.